src/proxy.c

```diff
@@ -191,30 +216,61 @@ gboolean match_rule(const json_t *rule,
 	gchar  *string;
 
 	json_entry = json_object_get(rule, entry);
+
+
 	if (!json_is_string (json_entry)) {
 		return FALSE;
 	}
 
 	string = (gchar*) json_string_value (json_entry);
-	if (verbose) {
-		g_print("will try matching %s with %s\n", string, comparison);
-        }
+	return compare_entry(comparison, string);
+}
 
-        if ((strcmp (string, "") != 0) &&
-                g_pattern_match_simple (string, comparison))
-        {
+/*! \brief Match a JSON method against an entry
+ *
+ * Compare a method as defined in the JSON rules file against the incoming or
+ * outgoing message. This is a helper function for is_allowed().
+ *
+ * \param  rule       The JSON rule
+ * \param  comparison The field to compare in the rule
+ * \return TRUE       if the rule matches the comparison
+ * \return FALSE      if there is no match
+ */
+gboolean match_method(const json_t *rule, const char *comparison)
+{
+    json_t *json_entry;
+    gchar  *string;
+
+    json_entry = json_object_get(rule, "method");
+    if (json_is_array(json_entry)) {
+        size_t ix;
+        json_t *val;
+
+        json_array_foreach(json_entry, ix, val) {
+            if (!json_is_string(val)) {
                 if (verbose) {
-                        g_print("was a match\n");
+                    g_print("Entry in method array is not a string.");
                 }
-                return TRUE;
-	} else {
+                return FALSE;
+            }
+
+            string = (gchar*) json_string_value (val);
+            if ((strcmp (string, "") != 0) && g_pattern_match_simple (string, comparison)) {
                 if (verbose) {
-                    g_print("no match\n");
+                    g_print("was a match\n");
                 }
-                return FALSE;
-	}
+                return TRUE;
+            }
+        }
+    } else if (json_is_string (json_entry)) {
+        string = (gchar*) json_string_value (json_entry);
+        return compare_entry(comparison, string);
+    }
+
+    return FALSE;
 }
 
+
```

Comment:
Throughout the code: remove the space between method and parenthesis, it's currently a mix of two styles. I know it might have been that already, but please be consistent with "no space" in your changes.